Meta/CMake/common_compile_options.cmake

```diff
@@ -50,8 +50,8 @@ macro(add_swift_link_options)
 endmacro()
 
 if (MSVC)
-    # On Windows, we always target the same architecture since there is no -march=native equivalent.
-    add_cxx_compile_options(/arch:AVX2)
+    # On Windows, compile for the native architecture of the host system.
+    add_cxx_compile_options(-march=native)
```

Comment:
This was meant as a default matching `ENABLE_CI_BASELINE_CPU`, which is now less the case by targeting the host system.

You could remove the `if (MSVC)` condition so it defaults to `-march=native` in the `else()`, and add a specific architecture in the `ENABLE_CI_BASELINE_CPU` case instead.